backend/src/schema/resolvers/follow.spec.js

```diff
@@ -1,36 +1,93 @@
-import { GraphQLClient } from 'graphql-request'
+import { createTestClient } from 'apollo-server-testing'
 import Factory from '../../seed/factories'
-import { host, login } from '../../jest/helpers'
+import { getDriver } from '../../bootstrap/neo4j'
+import createServer from '../../server'
+import { gql } from '../../jest/helpers'
 
 const factory = Factory()
-let clientUser1
-let headersUser1
+const driver = getDriver()
 
-const mutationFollowUser = id => `
-  mutation {
-    follow(id: "${id}", type: User)
+let query
+let mutate
+let authenticatedUser
+
+let user1
+let user2
+let variables
+
+const mutationFollowUser = gql`
+  mutation($id: ID!, $type: FollowTypeEnum) {
+    follow(id: $id, type: $type) {
+      name
+      followedBy {
+        id
+        name
+      }
+      followedByCurrentUser
+    }
   }
 `
-const mutationUnfollowUser = id => `
-  mutation {
-    unfollow(id: "${id}", type: User)
+
+const mutationUnfollowUser = gql`
+  mutation($id: ID!, $type: FollowTypeEnum) {
+    unfollow(id: $id, type: $type) {
+      name
+      followedBy {
+        id
+        name
+      }
+      followedByCurrentUser
+    }
   }
 `
 
-beforeEach(async () => {
-  await factory.create('User', {
-    id: 'u1',
-    email: 'test@example.org',
-    password: '1234',
-  })
-  await factory.create('User', {
-    id: 'u2',
-    email: 'test2@example.org',
-    password: '1234',
+const userQuery = gql`
+  query($id: ID) {
+    User(id: $id) {
+      followedBy {
+        id
+      }
+      followedByCurrentUser
+    }
+  }
+`
+
+beforeAll(() => {
+  const { server } = createServer({
+    context: () => ({
+      driver,
+      user: authenticatedUser,
+      cypherParams: {
+        currentUserId: authenticatedUser ? authenticatedUser.id : null,
+      },
+    }),
   })
 
-  headersUser1 = await login({ email: 'test@example.org', password: '1234' })
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-16T19:40:48Z" title="Monday, September 16th 2019, 9:40:48 pm +02:00">Sep 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -41,44 +70,57 @@ describe('follow', () => {
   describe('follow user', () => {
     describe('unauthenticated follow', () => {
       it('throws authorization error', async () => {
-        const client = new GraphQLClient(host)
-        await expect(client.request(mutationFollowUser('u2'))).rejects.toThrow('Not Authorised')
+        authenticatedUser = null
+        const { errors } = await mutate({
+          mutation: mutationFollowUser('u2'),
```


```suggestion
          mutation: mutationFollowUser, variables
```
Make the `mutationFollowUser` a `const` instead of a method. The string interpolation originates from a time when we weren't familiar enough with GraphQL and we did not know how to use `variables`.

The pattern which I happen to love is to mutate a global `variables` object in a `beforeEach` of a scope. Then pass these `variables` into the call of `mutate` or `query`. In my opinion that's a good way to avoid false negatives.

Reply:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-09-17T08:31:30Z" title="Tuesday, September 17th 2019, 10:31:30 am +02:00">Sep 17, 2019</time>_

---


done